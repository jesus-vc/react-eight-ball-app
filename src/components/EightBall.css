.EightBall-circle {
    width: 300px;
    height: 300px;
    border-radius: 50%; 
    text-align: center;    /* Center the text horizontally */
    line-height: 300px;    /* Center the text vertically (adjust based on circle size) */
    color: white;        
    font-weight: bold;
    font-size: x-large;
    
  }

.EightBall-button {
  color: black;        
  font-weight: bold;
  font-size: large;

}