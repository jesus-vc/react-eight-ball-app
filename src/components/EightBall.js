import "./EightBall.css";
import { useState } from "react";

const EightBall = ({ answers }) => {
  const initialState = {
    color: "black",
    msg: "Welcome! Think of a Question!",
    counters: 0,
  };

  const [color, setColor] = useState(initialState.color);
  const [msg, setMsg] = useState(initialState.msg);
  const [counters, setCounters] = useState({
    green: initialState.counters,
    goldenrod: initialState.counters,
    red: initialState.counters,
  });

  const setColorAndMsg = (answers) => {
    const randomIndex = Math.floor(Math.random() * answers.length);
    const randomColor = answers[randomIndex].color;
    const randomMsg = answers[randomIndex].msg;

    setColor(randomColor);
    setMsg(randomMsg);

    // create a shallow copy of the state with desired change
    setCounters((prevCounters) => ({
      ...prevCounters,
      [randomColor]: prevCounters[randomColor] + 1,
    }));
  };

  const resetState = () => {
    setColor(initialState.color);
    setMsg(initialState.msg);

    setCounters({
      green: initialState.counters,
      goldenrod: initialState.counters,
      red: initialState.counters,
    });
  };

  return (
    <>
      <div
        className="EightBall-circle"
        style={{ backgroundColor: color }}
        onClick={() => setColorAndMsg(answers)}
      >
        <p className="EightBall-msg">{msg}</p>
      </div>
      <button className="EightBall-button" onClick={resetState}>
        Reset
      </button>

      {Object.entries(counters).map(([color, count]) => (
        <p key={color} className="EightBall-counters">
          {`${color.charAt(0).toUpperCase() + color.slice(1)} Count: ${count}`}
        </p>
      ))}
    </>
  );
};

export default EightBall;
