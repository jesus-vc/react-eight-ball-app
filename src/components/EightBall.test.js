import { render, screen, fireEvent } from "@testing-library/react";
// import App from "./App";
import EightBall from "./EightBall";
import answers from "../data/answersData";

// smoke test
it("renders without crashing", () => {
  render(<EightBall answers={answers}> </EightBall>);
});

// snapshot test

it("matches snapshot", () => {
  const { asFragment } = render(<EightBall answers={answers}> </EightBall>);
  expect(asFragment()).toMatchSnapshot();
});

it("renders welcome text", () => {
  // const { getByText } = render(<EightBall answers={answers}> </EightBall>);
  // console.log(getByText("Welcome! Think of a Question!", { exact: true }));
  render(<EightBall answers={answers}> </EightBall>);
  expect(screen.getByText("Welcome! Think of a Question!")).toBeInTheDocument();
});

it("renders new text after click", () => {
  render(<EightBall answers={answers}> </EightBall>);

  const ball = screen.getByText("Welcome! Think of a Question!");
  expect(ball).toHaveTextContent("Welcome! Thinks of a Question!");
  debugger;
  fireEvent.click(ball);

  expect(ball).not.toHaveTextContent("Welcome! Think of a Question!");
  // expect(
  //   screen.queryByText("Welcome! Think of a Question!")
  // ).not.toBeInTheDocument();
});
