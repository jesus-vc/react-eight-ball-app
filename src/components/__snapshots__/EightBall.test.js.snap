// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="EightBall-circle"
    style="background-color: black;"
  >
    <p
      class="EightBall-msg"
    >
      Welcome! Think of a Question!
    </p>
  </div>
  <button
    class="EightBall-button"
  >
    Reset
  </button>
  <p
    class="EightBall-counters"
  >
    Green Count: 0
  </p>
  <p
    class="EightBall-counters"
  >
    Goldenrod Count: 0
  </p>
  <p
    class="EightBall-counters"
  >
    Red Count: 0
  </p>
</DocumentFragment>
`;
